File: Codechef/Starters-173/Cool_Subsequences.cpp
#include<bits/stdc++.h>
using namespace std;

void find_cool_subsequence(int n, vector<int>& arr) {
    unordered_set<int> seen;

    for (int num : arr) {
        if (seen.count(num)) {
            cout << 1 << endl;
            cout << num << endl;
            return;
        }
        seen.insert(num);
    }

    cout << -1 << endl;
}

int main() {
    int t;
    cin >> t;

    while (t--) {
        int n;
        cin >> n;
        vector<int> arr(n);

        for (int i = 0; i < n; i++) {
            cin >> arr[i];
        }

        find_cool_subsequence(n, arr);
    }

    return 0;
}

File: Codechef/Starters-173/Cool_Subsequences.py
import sys

def find_cool_subsequence(n, arr):
    seen = set()

    for num in arr:
        if num in seen:
            print(1)
            print(num)
            return
        seen.add(num)

    print(-1)

t = int(sys.stdin.readline().strip())

for _ in range(t):
    n = int(sys.stdin.readline().strip())
    arr = list(map(int, sys.stdin.readline().split()))
    find_cool_subsequence(n, arr)
